spring-cloud-deployer-spi-test/src/main/java/org/springframework/cloud/deployer/spi/test/AbstractTaskLauncherIntegrationTests.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.spi.test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.junit.Test;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.deployer.resource.maven.MavenProperties;
+import org.springframework.cloud.deployer.resource.maven.MavenResource;
+import org.springframework.cloud.deployer.spi.task.LaunchState;
+import org.springframework.cloud.deployer.spi.task.TaskLauncher;
+import org.springframework.cloud.deployer.spi.task.TaskStatus;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+
+/**
+ * Abstract base class for integration tests of
+ * {@link org.springframework.cloud.deployer.spi.task.TaskLauncher} implementations.
+ * <p>
+ * Inheritors should setup an environment with a newly created
+ * {@link org.springframework.cloud.deployer.spi.task.TaskLauncher}.
+ *
+ * Tests in this class are independent and leave the
+ * launcher in a clean state after they successfully run.
+ * </p>
+ * <p>
+ * As deploying a task is often quite time consuming, some tests assert
+ * various aspects of deployment in a row, to avoid re-deploying apps over and
+ * over again.
+ * </p>
+ *
+ * @author Eric Bottard
+ */
+public abstract class AbstractTaskLauncherIntegrationTests extends AbstractIntegrationTests {
+
+
+	protected abstract TaskLauncher taskLauncher();
+
+	@Autowired(required = false)
+	private MavenProperties mavenProperties;
+
+	@Test
+	public void testNonExistentAppsStatus() {
+		assertThat(taskLauncher().status(randomName()).getState(), is(LaunchState.unknown));
+	}
+
+
+
+
+
+	/**
+	 * Return a resource corresponding to the spring-cloud-deployer-spi-test-app app suitable for the target runtime.
+	 *
+	 * The default implementation returns an uber-jar fetched via Maven. Subclasses may override.
+	 */
+	protected Resource integrationTestProcessor() {
```

Comment:
That will need to be replaced by a task-specific equivalent.

Also, ideally the app-specific artifact should change name (as would this method), but this is a bit disruptive.
Hopefully `@Override` helps here..